### PR TITLE
APPS-1275 Implement retry logic in restore operation

### PR DESCRIPTION
**Pull request by korotkov-aerospike**

### CODE REVIEW

File: pkg/model/retry_policy.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	MaxRetries int `json:"maxRetries" yaml:"maxRetries"`
+}
+
+// Validate checks if the RetryPolicy fields are valid.
+func (rp *RetryPolicy) Validate() error {
+	if rp == nil {
+		return nil
+	}
+
+	if rp.BaseTimeout <= 0 {
```
Comment (filkeith) [inline pkg/model/retry_policy.go:29]: Why not uint64? if it must be positive

File: pkg/model/retry_policy.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		return fmt.Errorf("BaseTimeout must be greater than 0")
+	}
+
+	if rp.Multiplier < 1 {
+		return fmt.Errorf("multiplier must be greater or equal than 1")
+	}
+
+	if rp.MaxRetries < 0 {
```
Comment (filkeith) [inline pkg/model/retry_policy.go:37]: same here uint?

**Review by filkeith — COMMENTED** [review]
do we still need `modules/schema` ? and modules at all

File: pkg/model/retry_policy.go (hunk 1 above)
Comment (korotkov-aerospike) [reply pkg/model/retry_policy.go:29]: This is DTO object.
What if user passes -1? It will be decoded into a really big int.

**Comment by korotkov-aerospike** [comment]
@filkeith we do need schema, it is used to fetch aerospike configuration.

**Review by reugn — APPROVED** [review]

**Pull request merged**